cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -245,5 +286,27 @@ private long getTimeSeconds(Map<String, String> properties, String option) {
       String timeString = properties.get(option);
       return (timeString == null) ? -1 : TimeUnit.MILLISECONDS.toSeconds(Long.valueOf(timeString));
     }
+
+    /**
+     * Returns an instance of {@link AppMetadataStore}.
+     */
+    private AppMetadataStore getAppMetadataStore(DatasetContext context) {
+      // TODO Find a way to access the appMetadataStore without copying code from the DefaultStore
+      try {
+        Table table = DatasetsUtil.getOrCreateDataset(context, datasetFramework, APP_META_INSTANCE_ID,
+                                                      Table.class.getName(), DatasetProperties.EMPTY);
+        AppMetadataStore appMetadataStore = new AppMetadataStore(table, cConf, upgradeComplete);
+        // If upgrade was not complete, check if it is and update boolean
+        if (!upgradeComplete.get()) {
+          boolean isUpgradeComplete = appMetadataStore.isUpgradeComplete(APP_VERSION_UPGRADE_KEY);
```

Comment:
@maochf @poornachandra do you know? This is from current code, which I just move it from parent class to here.